pkg/cmd/repo/clone/clone.go

```diff
@@ -78,6 +79,10 @@ func NewCmdClone(f *cmdutil.Factory, runF func(*CloneOptions) error) *cobra.Comm
 }
 
 func cloneRun(opts *CloneOptions) error {
+	retryBackoffMinimum := time.Duration(1) * time.Millisecond
+	retryBackoff := time.Duration(100) * time.Millisecond
+	retryLimit := 3
```

Comment:
Even though implementing a retry with backoff from scratch in Go is a nice exercise, you could also consider using a 3rd party library. One such library already in use in our codebase seems to be used in Codespaces code: https://github.com/cli/cli/blob/79a1be4fe05f95a0f2b193f531ea91313668707f/internal/codespaces/codespaces.go#L48-L52

Throughout the gh codebase, we definitely have ad-hoc implementations for this in too many places, and the implementations are not even that good. Example: https://github.com/cli/cli/blob/79a1be4fe05f95a0f2b193f531ea91313668707f/pkg/cmd/pr/create/create.go#L753-L757

I wish I could point you to a better example on how to approach this, sorry 😅

Reply:
I copied my thing from somewhere. I'm not attached to any of the code. But it'd definitely be better if the codebase were consistent. I'm definitely not signing up to do that (I have a day job, and a side project, and I'm poking 3 other projects besides). -- Filed #6985 for someone else to pick up.

I'm happy to use `backoff`.

Reply:
Understood. You are welcome to move your implementation as-is to apply to the `repo fork` command and we can take it from there. Thanks for opening an additional housekeeping issue. 🙇

Reply:
I've moved it over and switched to the `ExponentialBackOff`, I don't mind those changes, as I have no particular attachment to things and that's basically a drop in replacement.